test/account/AccountPermissionHooks.t.sol

```diff
@@ -0,0 +1,239 @@
+                requireUOContext: false
+            })
+        );
+    }
+
+    // /// @dev Plugin 1 hook pair: [null, 2]
+    // ///      Expected execution: [null, 2]
+    // function test_postOnlyExecHook_run() public {
+    //     test_postOnlyExecHook_install();
+
+    //     vm.expectEmit(true, true, true, true);
+    //     emit ReceivedCall(
+    //         abi.encodeCall(IExecutionHook.postExecutionHook, (_POST_HOOK_FUNCTION_ID, "")),
+    //         0 // msg value in call to plugin
+    //     );
+
+    //     (bool success,) = address(account1).call(abi.encode(_EXEC_SELECTOR));
+    //     assertTrue(success);
+    // }
+
+    // function test_requireUOContextHook() public {
+    //     _installPlugin1WithHooks(
+    //         ManifestPermissionHook({
+    //             validationFunction: validationFunction,
+    //             functionId: _POST_HOOK_FUNCTION_ID,
+    //             isPreHook: false,
+    //             isPostHook: true,
+    //             requireUOContext: true
+    //         })
+    //     );
+
+    //     // Call should fail during validation!
+    //     PackedUserOperation memory uo = PackedUserOperation({
+    //         sender: address(this),
+    //         nonce: 0,
+    //         initCode: bytes(""),
+    //         callData: abi.encode(_EXEC_SELECTOR),
+    //         accountGasLimits: 0,
+    //         preVerificationGas: 0,
+    //         gasFees: 0,
+    //         paymasterAndData: bytes(""),
+    //         signature: bytes("")
+    //     });
+
+    //     vm.prank(address(entryPoint));
+    //     (bool success, bytes memory errMsg) =
+    //         address(account1).call(abi.encodeCall(BaseAccount.validateUserOp, (uo, bytes32(0), 0)));
+    //     assertFalse(success);
+    //     assertEq(bytes4(errMsg), IEntryPoint.FailedOpWithRevert.selector);
+    // }
+
+    function _installPlugin1WithHooks(ManifestPermissionHook memory permissionHooks) internal {
+        _m1.executionFunctions.push(
+            ManifestExecutionFunction({
+                executionSelector: _EXEC_SELECTOR,
+                isPublic: false,
+                allowSharedValidation: false
+            })
+        );
+        _m1.validationFunctions.push(validation);
+        _m1.permissionHooks.push(permissionHooks);
+        mockPlugin1 = new MockPlugin(_m1);
+        manifestHash1 = keccak256(abi.encode(mockPlugin1.pluginManifest()));
+
+        vm.expectEmit(true, true, true, true);
+        emit ReceivedCall(abi.encodeCall(IPlugin.onInstall, (bytes(""))), 0);
+        vm.expectEmit(true, true, true, true);
+        emit PluginInstalled(address(mockPlugin1), manifestHash1, new FunctionReference[](0));
+
+        vm.prank(address(entryPoint));
+        account1.installPlugin({
+            plugin: address(mockPlugin1),
+            manifestHash: manifestHash1,
+            pluginInstallData: bytes(""),
+            dependencies: new FunctionReference[](0)
+        });
+    }
+
+    fallback() external {}
```

Comment:
linter warning of non-payable fallback here

Reply:
removed